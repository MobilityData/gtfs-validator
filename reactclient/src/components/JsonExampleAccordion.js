/*
 *  Copyright (c) 2020. MobilityData IO.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import React, {Component} from 'react'
import {Accordion, Icon} from 'semantic-ui-react'
import JsonBeautyfier from "./JsonBeautyfier";
import localJson from "../data/json-examples/local-example.json";
import networkJson from "../data/json-examples/network-example.json";

export default class JsonExampleAccordion extends Component {
    state = {activeItemIndex: 0}

    handleClick = (e, titleProps) => {
        const {index} = titleProps
        const {activeItemIndex} = this.state
        const newIndex = activeItemIndex === index ? -1 : index

        this.setState({activeIndex: newIndex})
    }

    render() {
        const {activeItemIndex} = this.state;
        return (
            <Accordion styled id="json-example" className="json-example-container">
                <Accordion.Title
                    active={activeItemIndex === 0}
                    index={0}
                    onClick={this.handleClick}
                    className="json-example-tile"
                >
                    <Icon name='dropdown'/>
                    Example of configuration file using URL to remote GTFS archive </Accordion.Title>
                <Accordion.Content active={activeItemIndex === 0}>
                    <JsonBeautyfier id="json-example" data={localJson} className="json-example"/>

                </Accordion.Content>

                <Accordion.Title
                    active={activeItemIndex === 1}
                    index={1}
                    onClick={this.handleClick}
                >
                    <Icon name='dropdown'/>
                    Example of configuration file using locally stored GTFS archive
                </Accordion.Title>
                <Accordion.Content active={activeItemIndex === 1}>
                    <JsonBeautyfier id="json-example" data={networkJson} className="json-example"/>
                </Accordion.Content>
            </Accordion>
        )
    }
}
