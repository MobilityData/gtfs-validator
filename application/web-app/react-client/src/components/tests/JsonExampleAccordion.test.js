/*
 *  Copyright (c) 2020. MobilityData IO.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import React from "react";
import {shallow} from "enzyme";
import JsonExampleAccordion from "../JsonExampleAccordion";
import renderer from 'react-test-renderer';
import localJson from "../../data/json-examples/local-example.json";
import JsonBeautyfier from "../JsonBeautyfier";
import networkJson from "../../data/json-examples/network-example.json";

describe('JsonExampleAccordion', () => {
    jest.mock('../JsonBeautyfier')
    it('should be defined', () => {
        expect(JsonExampleAccordion).toBeDefined();
    });

    it('should render correctly', () => {
        const tree = renderer.create(
            <JsonExampleAccordion />
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should have two examples'", () => {
        jest.mock('../JsonBeautyfier')
        const tree = shallow(
            <JsonExampleAccordion />
        )
        // let mockFirstExample = mock/
        const firstExample = <JsonBeautyfier id="json-example" data={localJson} className="json-example"/>
        const secondExample = <JsonBeautyfier id="json-example" data={networkJson} className="json-example"/>;

        expect(tree.contains(firstExample)).toBe(true);
        expect(tree.contains(secondExample)).toBe(true);
    })

    //FIXME: adapt test to verify that function is called anytime time item is clicked on
    // it("should call function when title item is clicked", () => {
    //     let mockOnClick = jest.fn();
    //     const tree = renderer.create(
    //         <JsonExampleAccordion />
    //     );
    //     tree.root.props.onClick();
    //     expect(mockOnClick).toHaveBeenCalled();
    // })
})
