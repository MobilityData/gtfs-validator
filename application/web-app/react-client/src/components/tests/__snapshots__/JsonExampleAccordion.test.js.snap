// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonExampleAccordion should render correctly 1`] = `
<div
  className="accordion ui styled json-example-container"
  id="json-example"
>
  <div
    className="title json-example-tile"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    Example of configuration file using URL to remote GTFS archive 
  </div>
  <div
    className="content"
  >
    <pre
      style={
        Object {
          "backgroundColor": "rgb(39, 40, 34)",
          "overflow": "auto",
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        {
      </span>
      <div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              extract
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(242, 152, 59)",
                }
              }
            >
              relative/extraction/path
            </span>
            "
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              output
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(242, 152, 59)",
                }
              }
            >
              relative/output/path
            </span>
            "
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              input
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(242, 152, 59)",
                }
              }
            >
              relative/path/to/zipped_dataset
            </span>
            "
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              exclude
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(242, 152, 59)",
                }
              }
            >
              agency.txt,routes.txt
            </span>
            "
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              abort_on_error
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            true
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              beautify
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            false
          </span>
          
        </div>
      </div>
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        
        }
      </span>
    </pre>
  </div>
  <div
    className="title"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    Example of configuration file using locally stored GTFS archive
  </div>
  <div
    className="content"
  >
    <pre
      style={
        Object {
          "backgroundColor": "rgb(39, 40, 34)",
          "overflow": "auto",
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        {
      </span>
      <div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              extract
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(242, 152, 59)",
                }
              }
            >
              relative/extraction/path
            </span>
            "
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              output
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(242, 152, 59)",
                }
              }
            >
              relative/output/path
            </span>
            "
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              url
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(242, 152, 59)",
                }
              }
            >
              url/to/dataset
            </span>
            "
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              exclude
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(242, 152, 59)",
                }
              }
            >
              agency.txt,routes.txt
            </span>
            "
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              abort_on_error
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            true
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            ,
          </span>
        </div>
        <div>
          <span>
              
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            "
            <span
              style={
                Object {
                  "color": "rgb(201, 70, 56)",
                }
              }
            >
              beautify
            </span>
            "
            <span
              style={
                Object {
                  "color": "rgb(129, 211, 204)",
                }
              }
            >
              :
            </span>
             
          </span>
          <span
            style={
              Object {
                "color": "rgb(129, 211, 204)",
              }
            }
          >
            false
          </span>
          
        </div>
      </div>
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        
        }
      </span>
    </pre>
  </div>
</div>
`;
